src/ocean/io/select/selector/EpollFdSanity.d

```diff
@@ -0,0 +1,127 @@
+/*******************************************************************************
+
+    As linux on 64 machines is using only up to 56 bits for the user address
+    space (see https://www.kernel.org/doc/Documentation/x86/x86_64/mm.txt) we
+    can use bits from at least 56 to 63 for stuffing in the least significant 8
+    bits of the fd into the epoll registration. This can be used to check if
+    the ISelectClient's registration triggering in epoll wait is made for the
+    previous usage of the ISelectClient (for the wrong fd). This is not 100%
+    accurate, as the file descriptors could be reused but it can be in general
+    this is the best we can to find out expired registrations that are not
+    unregistered from epoll.
+
+    This module provides FdObjEpollData structure that binds fd
+    and the object used at the registration time, so that they both can
+    be registered with epoll and returned with epoll_wait to the user.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic labs GmbH.
+        All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.io.select.selector.EpollFdSanity;
+
+version (UnitTest)
+{
+    import ocean.core.Test;
+}
+
+/*******************************************************************************
+
+    Struct binding fd and a Object.
+
+*******************************************************************************/
+
+public struct FdObjEpollData
+{
+    /// Mask for the user-space address bits.
+    private const mask = 0x00FF_FFFF_FFFF_FFFF;
```

Comment:
Small suggestion:
```D
const address_bits = 56;
const address_mask = (2 ^^ (address_bits + 1)) - 1;
const fd_mask = ~address_mask;
```
Then you could remove the two instances of `56` and the `~mask` below.

Reply:
`static assert(address_mask + fd_mask == ulong.max);` :)